src/layer_normalization.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+import {add, sub, mul, div, pow} from './binary.js';
+import {reduceMean} from './reduce.js';
+import {reshape} from './reshape.js';
+import {Tensor, Scalar} from './lib/tensor.js';
+import {validateLayerNormalizationParams} from './lib/validate-input.js';
+
+/**
+ * Normalize the tensor values of input features using
+ * [layer-Normalization]
+ * @param {Tensor} input
+ * @param {Array} axes
+ * @param {MLBatchNormalizationOptions} [options]
+ * @return {Tensor}
+ */
+export function layerNormalization(input, {scale, bias, axes, epsilon=1e-5}) {
+  validateLayerNormalizationParams(...arguments);
+  console.log('axes :', axes);
+  if (axes === undefined) {
+    axes = Array.from({length: input.rank - 1}, (_, index) => index + 1);
+  }
+  // The output tensor has the same shape as the input tensor.
+  let output = new Tensor(input.shape);
+  const inputShape = input.shape;
+  const shape = new Array(input.rank).fill(1);
+  for (let i = 0; i < axes.length; i++) {
+    const axis = axes[i];
+    shape[axis] = inputShape[axis];
+  }
+  console.log('shape:', shape);
+  const reduceOptions = {axes, keepDimensions: true};
+  const mean = reduceMean(input, reduceOptions);
+  const variance = reduceMean(pow(sub(input, mean), new Scalar(2)), reduceOptions);
+  output = div(sub(input, mean),
+      pow(add(variance, new Scalar(epsilon)), new Scalar(0.5)));
```

Comment:
suggestion: **sqrt** op was added, so here we could invoke **sqrt** for simple usage.

Reply:
ok